### PR TITLE
feat(replay): Add beta badge for mobile replays

### LINKED ISSUE #73290: Add `beta` label to the Replay Mobile experience
We're closing in on our [Session Replay for Mobile - Open Beta](https://github.com/getsentry/sentry/issues/70065).

In order to help clarify this is till a beta offering, we'd like to add a `beta` label to the product.

# The label

image
The label when hovered should show a blurb with a link to the docs. A suggestion for the copy is:

> Session Replay for mobile apps is currently in beta. Beta features are still in progress and may have bugs. Learn more in [docs](TODO).

The link to the docs will clarify the mobile replay support and beta caveats

# Where to add it

The most important page to have this label is the Replay details page *for mobile*. It could live next to the page title, for example:

image

Additionally, on the Replay Index page, **when all projects selected are mobile projects**, we go into a "mobile view" mode, where the selector cards are hidden. In that case, we can also show the beta label on that page, example:

image

### PR DESCRIPTION
**Pull request by c298lee**
In replay list page if all mobile projects:
image
In replay details page if mobile replay:
image

Closes https://github.com/getsentry/sentry/issues/73290

### CODE REVIEW

File: static/app/components/replays/header/detailsPageBreadcrumbs.tsx [hunk 1]
```diff
@@ -52,7 +54,17 @@ function DetailsPageBreadcrumbs({orgSlug, replayRecord}: Props) {
           ) : null,
         },
         {
-          label: labelTitle,
+          label: isVideoReplay ? (
+            <span>
+              {labelTitle}
+              <FeatureBadge
+                type="beta"
+                title="Session Replay for mobile apps is currently in beta. Beta features are still in progress and may have bugs."
```
Comment (michellewzhang) [inline static/app/components/replays/header/detailsPageBreadcrumbs.tsx:62]:
```suggestion
                title={t('Session Replay for mobile apps is currently in beta. Beta features are still in progress and may have bugs.')}
```

File: static/app/views/replays/list.tsx [hunk 2]
```diff
@@ -34,6 +37,12 @@ function ReplaysListContainer() {
               )}
               docsUrl="https://docs.sentry.io/product/session-replay/"
             />
+            {allMobileProj ? (
+              <FeatureBadge
+                type="beta"
+                title="Session Replay for mobile apps is currently in beta. Beta features are still in progress and may have bugs."
```
Comment (michellewzhang) [inline static/app/views/replays/list.tsx:43]:
```suggestion
                title={t('Session Replay for mobile apps is currently in beta. Beta features are still in progress and may have bugs.')}
```

**Review by michellewzhang — APPROVED** [review]
loading looks good. could we make this `span` center aligned?

SCR-20240625-izsv

File: static/app/components/replays/header/detailsPageBreadcrumbs.tsx [hunk 3]
```diff
@@ -52,7 +54,19 @@ function DetailsPageBreadcrumbs({orgSlug, replayRecord}: Props) {
...
         },
         {
-          label: labelTitle,
+          label: isVideoReplay ? (
+            <span>
+              {labelTitle}
+              <FeatureBadge
+                type="beta"
+                title={t(
+                  'Session Replay for mobile apps is currently in beta. Beta features are still in progress and may have bugs.'
```
Comment (bruno-garcia) [inline static/app/components/replays/header/detailsPageBreadcrumbs.tsx:63]: The part about 'we recognize the irony' does fit?

Comment (bruno-garcia) [reply static/app/components/replays/header/detailsPageBreadcrumbs.tsx:63]: Oh @jas-kas may have rewritten it

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**